dialogue-core/src/main/java/com/palantir/dialogue/core/BlacklistingChannel.java

```diff
@@ -166,43 +136,59 @@ public String toString() {
 
     private final class BlacklistState {
         private final AtomicBoolean inProbation = new AtomicBoolean();
```

Comment:
why do we need a boolean and the `Probation` field - wouldn't an AtomicReference<Probation> or maybe volatile Probation do the trick using null or Optional.empty to signify that we're not in probation?

Reply:
Updated to use an AtomicReference for probation